File: getStockPrices.py

import pandas.io.data as web
import datetime as dt
import numpy as np
from pandas import Series, DataFrame, Panel
import matplotlib as mpl
import matplotlib.pyplot as plt
import pylab as pl


start = dt.datetime(2013,10,01)
end   = dt.datetime(2013,10,17)

goo = web.DataReader("GOOG", "yahoo", start, end)
yah = web.DataReader("YHOO", "yahoo", start, end)
gms = web.DataReader("GS",   "yahoo", start, end)
mgs = web.DataReader("MS",   "yahoo", start, end)
ibm = web.DataReader("IBM",  "yahoo", start, end)


plt.figure(1)
plt.subplot(121)
plt.plot(goo.index, goo.Close)
plt.title('Google Stock Price for 2013', fontsize=10)
plt.xlabel('Days', fontsize=6, color='red')
plt.ylabel('Closing Price', fontsize=6, color='red')
pl.tick_params(axis='y', which='major', labelsize=10)
pl.tick_params(axis='x', which='major', labelsize=10)
import matplotlib.figure.Figure as fig
fig.autofmt_xdate()


plt.subplot(122)
plt.plot(yah.index, yah["Close"], 'k.')
plt.title('Yahoo Stock Price for 2013', fontsize=10)
plt.xlabel('Days', fontsize=10, color='red')
plt.ylabel('Closing Price', fontsize=10, color='red')
pl.tick_params(axis='y', which='major', labelsize=8)
pl.tick_params(axis='x', which='major', labelsize=8)


plt.figure(2)
plt.subplot(121)
plt.plot(gms.index, gms["Close"], 'k.')
plt.title('Goldman Sachs Stock Price for 2013', fontsize=10)
plt.xlabel('Days', fontsize=10, color='red')
plt.ylabel('Closing Price', fontsize=10, color='red')
pl.tick_params(axis='y', which='major', labelsize=8)
pl.tick_params(axis='x', which='major', labelsize=8)


plt.subplot(122)
plt.plot(mgs.index, mgs["Close"], 'k.')
plt.title('Morgan Stanley Stock Price for 2013', fontsize=10)
plt.xlabel('Days', fontsize=10, color='red')
plt.ylabel('Closing Price', fontsize=10, color='red')
pl.tick_params(axis='y', which='major', labelsize=8)
pl.tick_params(axis='x', which='major', labelsize=8)


plt.figure(3)
plt.subplot(121)
plt.plot(ibm.index, ibm["Close"], 'k.')
plt.title('IBM Stock Price for 2013', fontsize=10)
plt.xlabel('Days', fontsize=10, color='red')
plt.ylabel('Closing Price', fontsize=10, color='red')
pl.tick_params(axis='y', which='major', labelsize=8)
pl.tick_params(axis='x', which='major', labelsize=8)


plt.figure(4)
plt.plot(goo.index, goo["Close"], 'b.', 
         yah.index, yah["Close"], 'g.', 
         gms.index, gms["Close"], 'r.', 
         mgs.index, mgs["Close"], 'c.', 
         ibm.index, ibm["Close"], 'y.')

plt.draw()
plt.show()




#bottom = plt.subplot2grid((4,4), (3,0), rowspan=1, colspan=4)
#bottom.bar(google.index, google['Volume'])
#plt.title('Google Trading Volume in Millions')

#plt.gcf().set_size_inches(15,8)




#import matplotlib.pyplot as plt
#top = plt.subplot2grid((4,4), (0, 0), rowspan=3, colspan=4)
#top.plot(goo.index, goo["Close"])
#plt.title('Google Stock Price for 2013')

#bottom = plt.subplot2grid((4,4), (3,0), rowspan=1, colspan=4)
#bottom.bar(google.index, google['Volume'])
#plt.title('Google Trading Volume in Millions')

#plt.gcf().set_size_inches(15,8)
#plt.show()


#close_pix.plt()


#from numpy import corrcoef, sum, log, arange
#from numpy.random import rand
#from pylab import pcolor, show, colorbar, xticks, yticks

# generating some uncorrelated data
#data = rand(10,100) # each row of represents a variable

# creating correlation between the variables
# variable 2 is correlated with all the other variables
#data[2,:] = sum(data,0)
# variable 4 is correlated with variable 8
#data[4,:] = log(data[8,:])*0.5

# plotting the correlation matrix
#R = corrcoef(data)
#pcolor(R)
#colorbar()
#yticks(arange(0.5,10.5),range(0,10))
#xticks(arange(0.5,10.5),range(0,10))
#plt.show()

File: getStockPrices.R
library(quantmod)
library(TTR)
library(gclus)


par(mfrow=c(3,2))


#The getSymbols function is used to retrieve stock data
symbols = c("GOOG","YHOO", "GS", "MS","IBM")
getSymbols(symbols) 


###################################################################
##Plot a subset of each company data seperately  
##
##Google 
chart_Series(GOOG, subset='2013', TA=NULL, name="Google Stock Price for 2013")
#addVo();
#addBBands();
#addCCI();
##Yahoo
chart_Series(YHOO, subset='2013', TA=NULL, name="Yahoo Stock Price for 2013")
##Goldman Sachs
chart_Series(GS,   subset='2013', TA=NULL, name="Goldman Sachs Stock Price for 2013")
##Morgan Stanley
chart_Series(MS,   subset='2013', TA=NULL, name="Morgan Stanley Stock Price for 2013")
##IBM
chart_Series(IBM,  subset='2013', TA=NULL, name="IBM Stock Price for 2013")


###################################################################
#Plot all company data together 
chart_Series(c(GOOG,YHOO,GS,MS,IBM), subset='2013', TA="", name=NULL)

#chart_Series(GOOG, subset='2013', TA="", name=NULL, clev=10)
#par(new=TRUE)
#chart_Series(YHOO, subset='2013', TA="", name=NULL)
#axis(side=2, tick=FALSE, labels=FLASE)
#par(new=TRUE)
#chart_Series(IBM, subset='2013', TA="", name=NULL)
#axis(side=2, labels=FLASE)


title("Stock Prices: Google, IBM, Goldman Sachs, Yahoo, and Morgan Stanley 
(in Descending Order)",  cex.main = 0.8,  font.main= 2, col.main= "blue")
mtext(text= "text1\ntext2\ntext3", side=1, cex = 0.75, font = 3, col="red",line=7)


###################################################################
#Plotting correlation data
a=GOOG['2013-10-02::2013-10-17']
b=YHOO['2013-10-02::2013-10-17']
c=  GS['2013-10-02::2013-10-17']
d=  MS['2013-10-02::2013-10-17']
e= IBM['2013-10-02::2013-10-17']

#Create a dataframe 
cn = 6
df = data.frame(a[,cn], b[,cn], c[,cn], d[,cn], e[,cn])
#df = data.frame(a$GOO.Open, b$YHOO.Open, c$GS.Open, d$MS.Open, e$IBM.Open)

colnames(df) = c("Google", "Yahoo", "GoldSach", "MorgStan", "IBM")
#pairs(~Google+Yahoo+GoldSach+MorgStan+IBM, data=df, pch=20, cex=0.25)
  
dim = length(df$Google)
cpairs(df, panel.colors=data.col, gap=.5, pch=20, cex=0.25, 
main="Variables Ordered and Colored by Correlation") 

print('                               ')
print('Printing correlation matrix ...')
corMat = cor(df)
print(corMat)
print('                               ')



#Use gclus library to get a color scatter plot
#data     <- df[c(1,2,3,4,5)]    # get data
#data.r   <- abs(cor(data))      # get correlations
data.col <- dmat.color(data.r)   # get colors
# Reorder variables so those with highest correlation are closest to the diagonal
#data.o   <- order.single(data.r)
#cpairs(data, data.o, panel.colors=data.col, gap=.5,
#main="Variables Colored by Correlation", pch=20, cex=0.25) 

#corMat = cor(df)
#image(x=seq(1,5), y=seq(1,5), z=corMat, 
#main="Variables Ordered and Colored by Correlation")
#text(expand.grid(x=seq(1,5), y=seq(1,5)), labels = round(c(corMat),2) )


#chartSeries(a, TA=NULL)
#barChart(a, theme='white.mono', bar.type='hlc', TA=NULL) 
#candleChart(a,multi.col=TRUE,theme='white', TA=NULL)
